api/src/main/java/jakarta/mail/Message.java

```diff
@@ -705,4 +707,33 @@ protected void setExpunged(boolean expunged) {
     public boolean match(SearchTerm term) throws MessagingException {
         return term.match(this);
     }
+
+    /**
+     * Obtains the {@link StreamProvider} from the session, if exists.
+     * Otherwise it obtains it from
+     * {@link Session#getDefaultInstance(java.util.Properties, Authenticator)}.
+     *
+     * @return the StreamProvider implementation from the session.
+     * @throws MessagingException if errors.
+     *
+     * @since JavaMail 2.2
+     */
+    protected StreamProvider provider() throws MessagingException {
```

Comment:
I used a short name because this method was private.  Now that it is protected maybe it should be `getStreamProvider` to match Session API?

Reply:
Yes, I was thinking the same. It is named `getStreamProvider` now.